expression/scalar_function.go

```diff
@@ -185,7 +185,15 @@ func newFunctionImpl(ctx sessionctx.Context, fold int, funcName string, retType
 	case ast.Cast:
 		return BuildCastFunction(ctx, args[0], retType), nil
 	case ast.GetVar:
-		return BuildGetVarFunction(ctx, args[0], retType)
+		sf, err := BuildGetVarFunction(ctx, args[0], retType)
+		if err != nil {
+			return nil, err
+		}
+		// GetVar try fold while fold = -1
```

Comment:
From the original comment of line#179 `-1 means try to fold constants if without errors/warnings `. But here we don't check whether there's warnings or not.
I think the fold can be simplified to a booo value?

Reply:
- The check warnings logic is in `tryFoldConstant` method.
- There is three situation 0,1,-1. So `fold` can not be bool.